host/sgx/create.c

```diff
@@ -317,13 +317,19 @@ static oe_result_t _calculate_enclave_size(
     control_size = 6 * OE_PAGE_SIZE;
 
     /* Compute end of the enclave */
+#ifdef OE_WITH_EXPERIMENTAL_EEID
+    size_t eeid_size = OE_PAGE_SIZE;
+
+    *enclave_end = image_size + eeid_size + heap_size +
+                   (size_settings->num_tcs * (stack_size + control_size));
+#else
     *enclave_end = image_size + heap_size +
                    (size_settings->num_tcs * (stack_size + control_size));
-
+#endif
     if (enclave_size)
     {
 #ifdef OE_WITH_EXPERIMENTAL_EEID
-        if (is_eeid_base_image(props))
+        if (is_eeid_base_image(&props->header.size_settings))
         {
             /* This is the maximum size SGX allows. When signing base images we
```

Comment:
SGX allows much bigger enclave linear address range. The comment should be updated. For exmaple, "Maxinum size allowed for EEID SGX enclave with heap/stack size and NumTCS set by the enclave loader".

Reply:
What does the enclave loader have to do with that? I found this number by trying to start enclaves with different sizes and this was the highest that my CPU didn't complain about. I'm sure newer CPUs will have other limits, but I haven't found any clear information about that yet.

Reply:
CPUID 12H sub-leaf0 (ECX=0) reports the max enclave size supported. Maybe some of the SGX CPUs do have 64GB as the max enclave size (64bit mode), but many other SGX CPUs have a larger limitation. I still think the comment should be changed. For exmaple "When signing base images we don't know the size that the final image will have, so we chose a size big enough to accomendate resonable heap/stack and #TCS selections, and supported by known SGX CPUs".

Reply:
I didn't know the CPU can tell us the max size, we should definitely do that and remove this comment.

Reply:
Actually, I'm wrong. In many use-cases images will be built on one CPU but run on another.